fastembed/image/transform/operators.py

```diff
@@ -75,52 +76,94 @@ def __call__(
-                crop_size = (crop_size, crop_size)
-            elif isinstance(crop_size, dict):
-                crop_size = (crop_size["height"], crop_size["width"])
+                Args:
+                    config (Dict[str, Any]): Configuration dictionary.
+
+                        Valid keys:
+                            - do_resize
+                            - size
+                            - do_center_crop
+                            - crop_size
+                            - do_rescale
+                            - rescale_factor
+                            - do_normalize
+                            - image_mean
+                            - image_std
+                        Valid size keys (nested):
+                            - {"height", "width"}
+                            - {"shortest_edge"}
+
+                Returns:
+                    Compose: Image processor.
+                """
+        transforms = []
+        cls._get_convert_to_rgb(transforms, config)
+        cls._get_resize(transforms, config)
+        cls._get_center_crop(transforms, config)
+        cls._get_pil2ndarray(transforms, config)
+        cls._get_rescale(transforms, config)
+        cls._get_normalize(transforms, config)
+        return cls(transforms=transforms)
+
+    @staticmethod
+    def _get_convert_to_rgb(transforms: List['Transform'], config: Dict[str, Any]):
+        transforms.append(ConvertToRGB())
+
+    @staticmethod
+    def _get_resize(transforms: List['Transform'], config: Dict[str, Any]):
+        mode = config.get('image_processor_type', 'CLIPImageProcessor')
+        if mode == 'CLIPImageProcessor':
+            if config.get("do_resize", False):
+                size = config["size"]
+                if "shortest_edge" in size:
+                    size = size["shortest_edge"]
+                elif "height" in size and "width" in size:
+                    size = (size["height"], size["width"])
+                else:
+                    raise ValueError("Size must contain either 'shortest_edge' or 'height' and 'width'.")
+                transforms.append(ClipResize(size=size, resample=config.get("resample", Image.Resampling.BICUBIC)))
+        elif mode == 'ConvNextFeatureExtractor':
+            if 'size' in config and "shortest_edge" not in config['size']:
+                raise ValueError(f"Size dictionary must contain 'shortest_edge' key. Got {config['size'].keys()}")
+            shortest_edge = config['size']["shortest_edge"]
+            crop_pct = config.get("crop_pct", 0.875)
+            if shortest_edge < 384:
+                # maintain same ratio, resizing shortest edge to shortest_edge/crop_pct
+                resize_shortest_edge = int(shortest_edge / crop_pct)
+                transforms.append(ClipResize(size=resize_shortest_edge, resample=config.get("resample", Image.Resampling.BICUBIC)))
+                transforms.append(CenterCrop(size=(shortest_edge, shortest_edge)))
```

Comment:
it's a temporary design, we would need to come up with something better, but since it is hidden from the users, it's okay to have it this way for now